src/kernels/kernelConnector.ts

```diff
@@ -87,7 +87,7 @@ export class KernelConnector {
         switch (selection) {
             case DataScience.restartKernel(): {
                 // Set our status
-                const status = statusProvider.set(DataScience.restartingKernelStatus());
+                const status = statusProvider?.set(DataScience.restartingKernelStatus());
```

Comment:
Why is this nullable? Can we mark this as nullable via `serviceContainer.get<IStatusProvider | undefined>(IStatusProvider);`
I don't think the type is nullable, hence tomorrow I might end up removing this as the type isn't nullable.

Reply:
It's not nullable, the get might fail. Oh I wonder if we have a tryGet? Maybe I should use that instead.

Reply:
Error handler should be there actually, I think I added that, but not 100% sure. But status provider isn't there at the moment. Ian actually found a bug related to that.

Reply:
The code in `src/platform/progress/statusProvider.ts` isn't use any node specific stuff, hence I don't see why we woudln't ever register this class and why it woudln't work in web. meaning it need not be nullable and we don't need to use `tryGet`.
Can we try to get that fixed instead of marking this nullable, as that feels like a bandaid fix.

Reply:
It is a bandaid fix, but I don't want to introduce the IStatusProvider at this late in the game. It's not necessary to fix IMO